src/product/product.service.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { validate as isUUID } from 'uuid';
+
+import { CreateProductDto } from './dto';
+import { Product } from './entities/product.entity';
+import { PaginationDto } from '../common/dto/pagination.dto';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<Product>,
+  ) {}
+
+  async findOne(searchTerm: string) {
+    let product: Product;
+
+    if (isUUID(searchTerm)) {
+      product = await this.productRepository.findOneBy({ id: searchTerm });
+    } else {
+      product = await this.productRepository.findOneBy({ name: searchTerm });
+    }
+
+    if (!product) throw new BadRequestException('Product not found');
```

Comment:
What do you think about changing this to a NotFoundException? Also, it could be a good approach to locate the error messages in a separate file, as @brunoBrizo suggested for now it could be part of your constants
```suggestion
    if (!product) throw new NotFoundException('Product not found');
```

Reply:
yes, I've made this change